src/hotspot/share/gc/shenandoah/shenandoahGeneration.cpp

```diff
@@ -221,7 +221,352 @@ void ShenandoahGeneration::prepare_gc(bool do_old_gc_bootstrap) {
   }
 }
 
-void  ShenandoahGeneration::prepare_regions_and_collection_set(bool concurrent) {
+void ShenandoahGeneration::compute_evacuation_budgets(ShenandoahHeap* heap, ShenandoahCollectionSet* collection_set,
```

Comment:
The refactoring of the large chunks of code into the smaller methods is definitely a step in the right direction in making the code easier to follow.

Nit: Would it make the code more readable to consolidate the long sections of comments into a block comment at the top of each of the methods and have much shorter comments in the body of the methods?

Reply:
Thanks for your suggestion.  Maybe you can do a subsequent pass (in a future PR) to clean up the structure of comments.  I'm sure these can be made better.  Sometimes, a "reader" of the comments has the best perspective on how to make the comments more valuable to other readers.